packages/analytics/dashboard-renderer/src/components/QueryDataProvider.vue

```diff
@@ -69,6 +78,43 @@ const { data, error, isValidating } = useSWRV(queryKey, async () => {
   }
 })
 
+// TODO: Support v4 natively and get rid of this hack.
+const data = computed(() => {
+  if (!v4Data.value) {
+    return undefined
+  }
+
+  const v4Meta = v4Data.value.meta
+
+  return {
+    records: v4Data.value.data.map(d => ({
+      ...d,
+      event: Object.entries(d.event).reduce((acc, [k, v]) => {
+        if (typeof v === 'string') {
+          acc[k.toUpperCase()] = v4Meta.display[k][v].name
+        } else {
+          acc[k.toUpperCase()] = v
+        }
+        return acc
+      }, {} as Record<string, any>),
```

Comment:
suggestion: some comments here and in the sections below might help a future-konger understand what's going on

Reply:
This is going away as soon as Filip comes back to https://konghq.atlassian.net/browse/MA-2480, so I'd prefer not to spend any time documenting it: in essence, getting rid of it is a blocker for releasing this chunk of work.